src/cli/src/meta_snapshot.rs

```diff
@@ -0,0 +1,145 @@
+    #[clap(long, default_value = "backup.bin")]
+    file_name: Option<String>,
+    /// The directory to store the snapshot file.
+    #[clap(long, default_value = ".")]
+    output_dir: Option<String>,
+    /// weather to use s3 as the output directory. default is false.
+    #[clap(long, default_value = "false")]
+    s3: bool,
+    /// The s3 bucket name.
+    #[clap(long)]
+    s3_bucket: Option<String>,
+    /// The s3 region.
+    #[clap(long)]
+    s3_region: Option<String>,
+    /// The s3 access key.
+    #[clap(long)]
+    s3_access_key: Option<String>,
+    /// The s3 secret key.
+    #[clap(long)]
+    s3_secret_key: Option<String>,
+    /// The s3 endpoint.
+    #[clap(long)]
+    s3_endpoint: Option<String>,
+    /// The s3 root path.
+    #[clap(long)]
+    s3_root: Option<String>,
+}
+
+impl MetaSnapshotCommand {
+    pub async fn build(&self) -> Result<Box<dyn Tool>, BoxedError> {
+        let max_txn_ops = self.max_txn_ops.unwrap_or(128);
+        let table_name = self
+            .meta_table_name
+            .as_ref()
+            .map(|n| n.as_str())
+            .unwrap_or(DEFAULT_META_TABLE_NAME);
+        let kvbackend = if let Some(etcd_addr) = &self.etcd_addr {
+            let addr = etcd_addr.split(',').collect::<Vec<_>>();
+            if addr.is_empty() {
+                return KVBackendNotSetSnafu { backend: "etcd" }
+                    .fail()
+                    .map_err(BoxedError::new);
+            }
+            EtcdStore::with_endpoints(addr, max_txn_ops)
+                .await
+                .map_err(BoxedError::new)?
+        } else if let Some(pg_addr) = &self.pg_addr {
+            // Initialize PgStore
+            // ...
+            PgStore::with_url(pg_addr, table_name, max_txn_ops)
+                .await
+                .map_err(BoxedError::new)?
+        } else if let Some(mysql_addr) = &self.mysql_addr {
+            MySqlStore::with_url(mysql_addr, table_name, max_txn_ops)
+                .await
+                .map_err(BoxedError::new)?
+        } else {
+            return KVBackendNotSetSnafu { backend: "etcd" }
+                .fail()
+                .map_err(BoxedError::new);
+        };
+
+        let output_dir = self.output_dir.clone().unwrap_or_else(|| ".".to_string());
+        let tool = if self.s3 {
+            let operator = ObjectStore::new(S3::default().root(&output_dir))
+                .context(OpenDalSnafu)
+                .map_err(BoxedError::new)?
+                .finish();
+            Box::new(MetaSnapshotTool {
+                inner: MetadataSnapshotManager::new(kvbackend, operator),
+            })
+        } else {
+            let operator = ObjectStore::new(Fs::default().root(&output_dir))
+                .context(OpenDalSnafu)
+                .map_err(BoxedError::new)?
+                .finish();
+            Box::new(MetaSnapshotTool {
+                inner: MetadataSnapshotManager::new(kvbackend, operator),
+            })
+        };
```

Comment:
It might be worth considering extracting the operator construction logic into a utility, as similar code exists in the import/export tool.